### PR TITLE
feat: add google cloud run starter workflows

### PR DESCRIPTION
**Pull request by verbanicm**
This PR adds starter workflows for Google Cloud Run.

---

## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [x] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [x] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least priviledge [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully. 

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [ ] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [ ] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

**Comment by bharathkkb** [comment]
@abirismyname @imjohnbo could you PTAL :)

**Comment by bishal-pdMSFT** [comment]
@anuragc617 @azooinmyluggage please review

**Review by azooinmyluggage — APPROVED** [review]
LGTM

**Comment by ashwinsangem** [comment]
@verbanicm, I have approved the run for you.

**Comment by verbanicm** [comment]
@ashwinsangem Thanks! Looks good now!

**Comment by asaneh** [comment]
> این روابط عمومی گردش‌های کاری آغازگر را برای Google Cloud Run اضافه می‌کند.
> 
> ## پیش نیازها
> * [x] قبل از ارسال یک گردش کار جدید، لطفاً برای پیوستن به برنامه شریک فناوری GitHub درخواست دهید: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner) .
> 
> ### **لطفاً توجه داشته باشید که در حال حاضر ما فقط گردش‌های کاری شروع جدید را برای اسکن کد می‌پذیریم. به‌روزرسانی‌های گردش کار آغازگر موجود خوب هستند.**
> ## وظایف
> **برای _همه_ گردش کار، گردش کار:**
> 
> * [x] باید در `.yml`فایلی با زبان یا پلتفرم به عنوان نام فایل آن، با فرمت [_کوچکتر و با حروف کباب_](https://en.wikipedia.org/wiki/Kebab_case) (به عنوان مثال، [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)) موجود باشد. نویسه‌های ویژه باید حذف شوند یا در صورت لزوم با کلمات جایگزین شوند (به عنوان مثال، "dotnet" به جای ".NET").
> * [x] باید از حروف جمله برای نام گردش کار و مراحل استفاده شود (به عنوان مثال، "آزمون های اجرا").
> * [x] باید _فقط_ با نام زبان یا پلتفرم نامگذاری شود (به عنوان مثال، "Go"، نه "Go CI" یا "Go Build").
> * [x] باید نظراتی را در گردش کار برای هر قسمتی که واضح نیست یا می تواند از توضیح استفاده کند، درج کند.
> * [x] باید حداقل [مجوزهای](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) مربوط `GITHUB_TOKEN`به آن را مشخص کنید تا گردش کار با موفقیت اجرا شود.
> 
> **برای گردش کار _CI_ ، گردش کار:**
> 
> * [ ] باید در زیر [فهرست نگهداری ](https://github.com/actions/starter-workflows/tree/main/ci)[شود`ci`](https://github.com/actions/starter-workflows/tree/main/ci) .
> * [ ] باید شامل یک فایل منطبق `ci/properties/*.properties.json`باشد (به عنوان مثال، [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
> * [ ] باید به `push`سمت `branches: [ $default-branch ]`و `pull_request`سوی اجرا شود `branches: [ $default-branch ]`.
> * [ ] گردش کار بسته بندی باید `release`با `types: [ created ]`.
> * [ ] گردش کار انتشار باید نام فایلی داشته باشد که نام زبان یا پلتفرم باشد، با حروف کوچک و پس از آن "-publish" (به عنوان مثال، [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).
> 
> **برای گردش کار _اسکن کد_ ، گردش کار:**
> 
> * [ ] باید در زیر [فهرست نگهداری ](https://github.com/actions/starter-workflows/tree/main/ci)[شود`code-scanning`](https://github.com/actions/starter-workflows/tree/main/ci) .
> * [ ] باید شامل یک فایل منطبق `code-scanning/properties/*.properties.json`(به عنوان مثال، [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json))، با ویژگی های تنظیم شده به صورت زیر باشد:
>   
>   * [ ]  `name`: نام ادغام کد اسکن.
>   * [ ]  `organization`: نام سازمان تولید کننده کد ادغام اسکن.
>   * [ ]  `description`: شرح کوتاهی از ادغام کد اسکن.
>   * [ ]  `categories`: آرایه ای از زبان های پشتیبانی شده توسط کد اسکن ادغام.
>   * [ ]  `iconName`: نام آرم SVG که نشان دهنده ادغام کد اسکن است. این آرم SVG باید در [فهرست `icons`موجود](https://github.com/actions/starter-workflows/tree/main/icons) باشد.
> * [ ] باید به `push`سمت `branches: [ $default-branch, $protected-branches ]`و `pull_request`سوی اجرا شود `branches: [ $default-branch ]`. ما همچنین یک `schedule`ماشه از `cron: $cron-weekly`(به عنوان مثال، [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)) را توصیه می کنیم.
> 
> **چند نکته کلی:**
> 
> * [x]  This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
> * [x]  This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
>   ```
>   # This workflow uses actions that are not certified by GitHub.
>   # They are provided by a third-party and are governed by
>   # separate terms of service, privacy policy, and support
>   # documentation.
>   ```
> * [ ]  Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
> * [ ]  Automation and CI workflows cannot be dependent on a paid service or product.

File: deployments/properties/google-cloudrun-docker.properties.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "Build and Deploy to Cloud Run",
+  "description": "Build a Docker container, publish it to Google Artifact Registry, and deploy to Google Cloud Run.",
+  "creator": "Google Cloud",
+  "iconName": "google-cloud",
+  "categories": ["Deployment", "Containers", "Dockerfile", "Cloud Run", "Serverless"]
```
Comment (bishal-pdMSFT) [inline deployments/properties/google-cloudrun-docker.properties.json:6]: @verbanicm do the categories of "Cloud run" and "Serverless" mean that there will be specific files for these in user's repo. If yes, then we can help enhance a repo analysis logic which we have today that detects certain files and prioritizes such workflows when showing to user.
Let me know if that is the case and I can help with more details on how to make such enhancement.

**Comment by winstonsmith132** [comment]
let know

Sent from ProtonMail mobile

-------- Original Message --------

**Pull request merged**